From 223c327e4b7ec01d5930834e2484c2f6fe689cf3 Mon Sep 17 00:00:00 2001
From: Matteo Geromin <geromin.matteo@gmail.com>
Date: Wed, 4 Nov 2015 22:50:50 +0100
Subject: [PATCH 08/15] Add shutdown command

---
 board/ti/am335x/board.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/board/ti/am335x/board.c b/board/ti/am335x/board.c
index 7926a95..e21cf00 100644
--- a/board/ti/am335x/board.c
+++ b/board/ti/am335x/board.c
@@ -9,6 +9,7 @@
  */
 
 #include <common.h>
+#include <command.h>
 #include <rtc.h>
 #include <errno.h>
 #include <spl.h>
@@ -903,3 +904,13 @@ int board_eth_init(bd_t *bis)
 	return n;
 }
 #endif
+
+static int do_shutdown(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[]) {
+	am335x_rtc_poweroff();
+	return 0;
+}
+
+U_BOOT_CMD(shutdown, 1, 0, do_shutdown,
+"shutdown \n",
+" - power off the board\n"
+);
-- 
1.9.1

