From 099d73919764e4fe1716a72e4d0e510454b480a3 Mon Sep 17 00:00:00 2001
From: Matteo Geromin <geromin.matteo@gmail.com>
Date: Thu, 7 Jan 2016 14:50:08 +0100
Subject: [PATCH 16/19] Add ramdisk size parameter on kernel bootcmd line

---
 include/configs/am335x_evm.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index f45e769..3f048d4 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -138,7 +138,8 @@ extern unsigned long fb_mem_baseaddr;
 	"ramargs=setenv bootargs console=${console} " \
 		"${optargs} " \
 		"root=${ramroot} " \
-		"rootfstype=${ramrootfstype}\0" \
+		"rootfstype=${ramrootfstype} " \
+		"ramdisk_size=32768\0" \
 	"loadramdisk=load mmc ${mmcdev} ${rdaddr} ${bootdir}/ramdisk-bbb.gz\0" \
 	"loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
 	"loadinitrd=load mmc ${bootpart} ${initrd_addr} ${bootdir}/${initrd_file}; setenv initrd_size ${filesize}\0" \
-- 
1.9.1

